{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "import sqlite3\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create gene table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "metadata": {},
   "outputs": [
    {
     "ename": "OperationalError",
     "evalue": "table genes already exists",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOperationalError\u001b[0m                          Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-293-2fce42b21159>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      9\u001b[0m                                 \u001b[0mstart\u001b[0m \u001b[0mINT\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m                                 \u001b[0mend\u001b[0m \u001b[0mINT\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 11\u001b[0;31m                                 strand VARCHAR(1))\"\"\")\n\u001b[0m",
      "\u001b[0;31mOperationalError\u001b[0m: table genes already exists"
     ]
    }
   ],
   "source": [
    "conn = sqlite3.connect('my.db')\n",
    "c = conn.cursor()\n",
    "\n",
    "#Gene table\n",
    "c.execute(\"\"\"CREATE TABLE genes (id INT PRIMARY KEY ASC, \n",
    "                                name TEXT, \n",
    "                                description TEXT, \n",
    "                                chromosome TEXT, \n",
    "                                start INT, \n",
    "                                end INT,\n",
    "                                strand VARCHAR(1))\"\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Insert into gene table:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Hexokinase gene; glycolysis in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "                                name,\n",
    "                                description,\n",
    "                                chromosome,\n",
    "                                start,\n",
    "                                end) \n",
    "                        VALUES(3098,\n",
    "                        'Homo sapiens hexokinase 1 (HK1), transcript variant 9',\n",
    "                        'Convert glucose to Glucose-6-Phosphate in glycolysis',\n",
    "                        'chr10',\n",
    "                        '102',\n",
    "                        '2759');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "\n",
    "\n",
    "#Drosophila melanogaster\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "                                name,\n",
    "                                description,\n",
    "                                chromosome,\n",
    "                                start,\n",
    "                                end) \n",
    "                        VALUES('45875',\n",
    "                        'Drosophila melanogaster hexokinase A, transcript variant A (Hex-A)',\n",
    "                        'Convert glucose to Glucose-6-Phosphate in glycolysis',\n",
    "                        'chr x',\n",
    "                        '824',\n",
    "                        '2170');\"\"\")\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "\n",
    "#E. coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "                                name,\n",
    "                                description,\n",
    "                                chromosome) \n",
    "                        VALUES('946858',\n",
    "                        'Escherichia coli (strain K12) Glucokinase',\n",
    "                        'Facilitates phosphorylation of glucose to glucose-6-phosphate',\n",
    "                        '-');\"\"\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Phosphofructokinase gene; glycolysis in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('5213',\n",
    "'Human phosphofructokinase (PFK)',\n",
    "'Convert Fructose-6-phosphate to Fructose-1,6-bisphosphate',\n",
    "'chr11',\n",
    "1,\n",
    "905);\"\"\")\n",
    "\n",
    "\n",
    "#Drosophila melanogaster\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('36060',\n",
    "'Drosophila melanogaster phosphofructokinase, transcript variant E(Pfk)',\n",
    "'Convert Fructose-6-phosphate to Fructose-1,6-bisphosphate',\n",
    "'chr2R',\n",
    "103,\n",
    "2871);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "\n",
    "#E. coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description)\n",
    "VALUES('946676',\n",
    "'1-phosphofructokinase',\n",
    "'Convert Fructose 6-phosphate to Fructose-1,6-bisphosphate');\"\"\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Enolase gene; glycolysis in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('X66610',\n",
    "'Homo sapiens enolase',\n",
    "'Convert 2-phosphoglycerate (2-PG) to phosphoenolpyruvate (PEP)',\n",
    "'chr17',\n",
    "40,\n",
    "146);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "#Drosophila melanogaster\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('NM_001169382',\n",
    "'Drosophila melanogaster enolase, transcript variant F (Eno)',\n",
    "'Convert 2-phosphoglycerate (2-PG) to phosphoenolpyruvate (PEP)',\n",
    "'chr2L',\n",
    "979,\n",
    "2280);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "\n",
    "#E.coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description)\n",
    "VALUES('945032',\n",
    "'Escherichia coli Enolase',\n",
    "'Catalyzes the reversible conversion of 2-phosphoglycerate into phosphoenolpyruvate');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pyruvate kinase gene; glycolysis in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('5315',\n",
    "'Homo sapiens gene for pyruvate kinase L',\n",
    "'Convert phosphoenolpyruvate to adenosine diphosphate,',\n",
    "'chr1',\n",
    "517,\n",
    "3688);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "#Drosophila melanogaster\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('42620',\n",
    "'Drosophila melanogaster pyruvate kinase, transcript variant A(PyK)',\n",
    "'Convert phosphoenolpyruvate to adenosine diphosphate,',\n",
    "'chr3R',\n",
    "343,\n",
    "1844);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "#E.coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description)\n",
    "VALUES('946179',\n",
    "'E. coli Pyruvate kinase I gene(pykF)',\n",
    "'Synthesizes pyruvate from D-glyceraldehyde 3-phosphate');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Aconitase gene; TCA in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('50',\n",
    "'Homo sapiens chromosome 22 aconitase (ACO2)',\n",
    "'Convert citrate to isocitrate',\n",
    "'chr22',\n",
    "274,\n",
    "8410);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "#Drosophila melanogaster\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('44149',\n",
    "'Drosophila melanogaster aconitase, transcript variant F (Acon)',\n",
    "'Convert citrate to isocitrate',\n",
    "'chr2L',\n",
    "203,\n",
    "2566);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "#E. coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('944864',\n",
    "'Escherichia coli (strain K12) Aconitate hydratase B',\n",
    "'Catalyzes the reversible isomerization of citrate to isocitrate via cis-aconitate',\n",
    "'-');\"\"\")\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Malate Dehydrogenase gene; TCA in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('4190',\n",
    "'Homo sapiens malate dehydrogenase 1 (MDH1)',\n",
    "'Catalyzes the reversible oxidation of malate to oxaloacetate',\n",
    "'chr2');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "#Drosophila melanogaster\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('34414',\n",
    "'Drosophila melanogaster malate dehydrogenase 1, transcript variant B (Mdh1)',\n",
    "'Catalyzes the reversible oxidation of malate to oxaloacetate',\n",
    "'chr2L');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "#E. coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('947854',\n",
    "'Escherichia coli (strain K12) Malate dehydrogenase(mdh)',\n",
    "'Catalyzes the reversible oxidation of malate to oxaloacetate',\n",
    "'-');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Isocitrate dehydrogenase gene; TCA in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('3417',\n",
    "'Human IDH1 isocitrate dehydrogenase (NADP(+)) 1',\n",
    "'catalyzes the oxidative decarboxylation of isocitrate',\n",
    "44,\n",
    "1303,\n",
    "'-');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "#Drosophila melanogaster \n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('44291',\n",
    "'Drosophila melanogaster isocitrate dehydrogenase, transcript variant K (Idh)',\n",
    "'catalyzes the oxidative decarboxylation of isocitrate',\n",
    "284,\n",
    "1723,\n",
    "'3L');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "#E.coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('945702',\n",
    "'Escherichia coli (strain K12) Isocitrate dehydrogenase [NADP] (icd)',\n",
    "'catalyzes the oxidative decarboxylation of isocitrate',\n",
    "'-');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pyruvate carboxylase gene; TCA in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('5091',\n",
    "'Homo sapiens pyruvate carboxylase(PC)',\n",
    "'carboxylation of pyruvate to form oxaloacetate',\n",
    "'chr11',\n",
    "84,\n",
    "3620);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())\n",
    "\n",
    "#Drosophila melanogaster\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('36020',\n",
    "'Drosophila melanogaster pyruvate carboxylase, transcript variant M(PCB)',\n",
    "'carboxylation of pyruvate to form oxaloacetate',\n",
    "'2R',\n",
    "347,\n",
    "3940);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())\n",
    "\n",
    "#E.coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('948457',\n",
    "'Escherichia coli (strain K12) Phosphoenolpyruvate carboxylase(ppc)',\n",
    "'Forms oxaloacetate, a four-carbon dicarboxylic acid source for the tricarboxylic acid cycle',\n",
    "'-');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## G6-P Dehydrogenase gene; Pentose phosphate pathway in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('2539',\n",
    "'Homo sapiens glucose-6-phosphate dehydrogenase (G6PD)',\n",
    "'Convert Glucose-6-Phosphate to 6-phosphogluconate',\n",
    "822,\n",
    "3824,\n",
    "'-');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())\n",
    "\n",
    "#Drosophila melanogaster\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('40445',\n",
    "'Drosophila melanogaster glucose-6-phosphate dehydrogenase (G6PD)',\n",
    "'Convert Glucose-6-Phosphate to 6-phosphogluconate',\n",
    "'chr X',\n",
    "438,\n",
    "2756);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())\n",
    "\n",
    "#E.coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('946370',\n",
    "'Escherichia coli (strain K12) Glucose-6-phosphate 1-dehydrogenase(zwf)',\n",
    "'Convert Glucose-6-Phosphate to 6-phosphogluconate',\n",
    "'-');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Phosphoglucomutase gene; Pentose phosphate pathway in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('5236',\n",
    "'Human phosphoglucomutase 1 (PGM1)',\n",
    "'Facilitates the interconversion of glucose 1-phosphate and glucose 6-phosphate',\n",
    "'chr1',\n",
    "63,\n",
    "1751);\"\"\")\n",
    "\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())\n",
    "\n",
    "#Drosophila melanogaster\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('44010',\n",
    "'Drosophila melanogaster phosphoglucomutase (Pgm)',\n",
    "'Facilitates the interconversion of glucose 1-phosphate and glucose 6-phosphate',\n",
    "'chr3L',\n",
    "28,\n",
    "1710);\"\"\")\n",
    "\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())\n",
    "\n",
    "#E.coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('945271',\n",
    "'Escherichia coli (strain K12) phosphoglucomutase (Pgm)',\n",
    "'Facilitates the interconversion of glucose 1-phosphate and glucose 6-phosphate',\n",
    "'-');\"\"\")\n",
    "\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Transaldolase gene; Pentose phosphate pathway in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('6888',\n",
    "'Homo sapiens transaldolase 1 (TALDO1)',\n",
    "'Links the pentose phosphate pathway to glycolysis',\n",
    "'chr11');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())\n",
    "\n",
    "#Drosophila melanogaster\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('37804',\n",
    "'Drosophila melanogaster transaldolase (Taldo)',\n",
    "'Links the pentose phosphate pathway to glycolysis',\n",
    "'chr2R',\n",
    "21,\n",
    "1016);\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "print(c.fetchone())\n",
    "\n",
    "#E. coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('947006',\n",
    "'Escherichia coli (strain K12) Transaldolase A',\n",
    "'Links the pentose phosphate pathway to glycolysis',\n",
    "'-');\"\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ribose 5-phosphate isomerase gene; Pentose phosphate pathway in human, Drosophila and E.coli"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Human\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('22934',\n",
    "'Homo sapiens ribose 5-phosphate isomerase (RPI)',\n",
    "'Catalyzes the reversible conversion between ribose-5-phosphate and ribulose-5-phosphate',\n",
    "'chr2',\n",
    "42,\n",
    "977);\"\"\")\n",
    "\n",
    "#Drosophila melanogaster \n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome,\n",
    "       start,\n",
    "       end)\n",
    "VALUES('246599',\n",
    "'Drosophila melanogaster Ribose-5-phosphate isomerase (Rpi)',\n",
    "'Catalyzes the reversible conversion between ribose-5-phosphate and ribulose-5-phosphate',\n",
    "'chr2',\n",
    "72,\n",
    "897);\"\"\")\n",
    "\n",
    "\n",
    "#E.coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "       name,\n",
    "       description,\n",
    "       chromosome)\n",
    "VALUES('948602',\n",
    "'Escherichia coli (strain K12) Ribose-5-phosphate isomerase B',\n",
    "'Catalyzes the reversible conversion between ribose-5-phosphate and ribulose-5-phosphate',\n",
    "'-');\"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pip install beautifultable\n",
    "from beautifultable import BeautifulTable\n",
    "table.column_headers = [\"id\",\"name\", \"description\", \"chromosome\",\"start\",\"end\"]\n",
    "table.append_row([\"3098\",\n",
    "                  'Homo sapiens hexokinase 1 (HK1), transcript variant 9',\n",
    "                  'Convert glucose to Glucose-6-Phosphate in glycolysis',\n",
    "                  'chr10',\n",
    "                  '102',\n",
    "                  '2759'])\n",
    "table.append_row([\"45875\",\n",
    "                  'Drosophila melanogaster hexokinase A, transcript variant A (Hex-A)',\n",
    "                  'Convert glucose to Glucose-6-Phosphate in glycolysis',\n",
    "                  'chr x',\n",
    "                  '824',\n",
    "                  '2170'])\n",
    "table.append_row([\"946858\",\n",
    "                  'Escherichia coli (strain K12) Glucokinase',\n",
    "                  'Facilitates phosphorylation of glucose to glucose-6-phosphate',\n",
    "                  '-'])\n",
    "print(table)\n",
    "\n",
    "\n",
    "#E. coli\n",
    "c.execute(\"\"\"INSERT INTO genes (id,\n",
    "                                name,\n",
    "                                description,\n",
    "                                chromosome) \n",
    "                        VALUES('946858',\n",
    "                        'Escherichia coli (strain K12) Glucokinase',\n",
    "                        'Facilitates phosphorylation of glucose to glucose-6-phosphate',\n",
    "                        '-');\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conn.commit\n",
    "c.execute(\"SELECT * from genes;\")\n",
    "\n",
    "print('Hexokinase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(\"\")\n",
    "\n",
    "print('Ribose 5-phosphate isomerase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(\"\")\n",
    "\n",
    "print('Phosphofructokinase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "\n",
    "print('Enolase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(\"\")\n",
    "\n",
    "print('Pyruvate kinase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(\"\")\n",
    "\n",
    "print('Aconitase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(\"\")\n",
    "\n",
    "print('Malate Dehydrogenase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(\"\")\n",
    "\n",
    "print('Isocitrate dehydrogenase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(\"\")\n",
    "\n",
    "print('Pyruvate carboxylase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(\"\")\n",
    "\n",
    "print('G6-P Dehydrogenase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(\"\")\n",
    "print('Phosphoglucomutase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(\"\")\n",
    "print('Transaldolase gene')\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pathway table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "c.execute(\"\"\"CREATE TABLE pathway (id INT, \n",
    "    name TEXT, \n",
    "    description TEXT);\"\"\")\n",
    "\n",
    "#Insert Glycolysis pathway\n",
    "c.execute(\"\"\"INSERT INTO pathway (id,\n",
    "       name,\n",
    "       description)\n",
    "VALUES('1',\n",
    "'Glycolysis',\n",
    "'Oxidative breakdown of one glucose into two pyruvate with the capture of some energy as ATP and NADH');\"\"\")\n",
    "\n",
    "\n",
    "#Insert TCA pathway\n",
    "c.execute(\"\"\"INSERT INTO pathway (id,\n",
    "       name,\n",
    "       description)\n",
    "VALUES('2',\n",
    "'TCA Cycle',\n",
    "'Form a six-carbon molecule that is used to produce NADH, carbon dioxide, ATP and FADH2');\"\"\")\n",
    "\n",
    "\n",
    "\n",
    "#Insert Pentose-phosphate pathway\n",
    "c.execute(\"\"\"INSERT INTO pathway (id,\n",
    "       name,\n",
    "       description)\n",
    "VALUES('3',\n",
    "'Pentose-Phosphate Pathway',\n",
    "'Generates NADPH and pentoses (5-carbon sugars) as well as ribose 5-phosphate, precursor for the synthesis of nucleotides');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from pathway;\")\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Enzyme table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create enzyme table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "c.execute(\"\"\"CREATE TABLE enzyme (id INT, \n",
    "    name TEXT, \n",
    "    function TEXT, \n",
    "    enzyme_commission_number TEXT);\"\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Insert into enzyme table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Hexokinase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "        name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('1',\n",
    "'Hexokinase',\n",
    "'Form a six-carbon molecule that is used to produce NADH, carbon dioxide, ATP and FADH2',\n",
    "'2.7.1.1');\"\"\")\n",
    "\n",
    "\n",
    "#Glucokinase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('2',\n",
    "'Escherichia coli (strain K12) Glucokinase',\n",
    "'Facilitates phosphorylation of glucose to glucose-6-phosphate',\n",
    "'2.7.1.2');\"\"\")\n",
    "\n",
    "\n",
    "#Phosphofructokinase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('3',\n",
    "'Phosphofructokinase',\n",
    "'Convert Fructose-6-phosphate to Fructose-1,6-bisphosphate',\n",
    "'2.7.1.11');\"\"\")\n",
    "\n",
    "\n",
    "#Phosphofructokinase of E.coli\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('4',\n",
    "'1-Phosphofructokinase',\n",
    "'Convert Fructose-6-phosphate to Fructose-1,6-bisphosphate',\n",
    "'2.7.1.56');\"\"\")\n",
    "\n",
    "\n",
    "#Enolase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('5',\n",
    "'Enolase',\n",
    "'Convert 2-phosphoglycerate (2-PG) to phosphoenolpyruvate (PEP)',\n",
    "'4.2.1.11');\"\"\")\n",
    "\n",
    "\n",
    "#Pyruvate kinase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('6',\n",
    "'Pyruvate kinase',\n",
    "'Convert 2-phosphoglycerate (2-PG) to phosphoenolpyruvate (PEP)',\n",
    "'2.7.1.40');\"\"\")\n",
    "\n",
    "\n",
    "#Aconitase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('7',\n",
    "'Aconitase',\n",
    "'Convert citrate to isocitrate',\n",
    "'4.2.1.3');\"\"\")\n",
    "\n",
    "\n",
    "\n",
    "#Malate dehydrogenase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('8',\n",
    "'Malate dehydrogenase',\n",
    "'Catalyzes the reversible oxidation of malate to oxaloacetate',\n",
    "'1.1.1.37');\"\"\")\n",
    "\n",
    "\n",
    "\n",
    "#Isocitrate dehydrogenase \n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('9',\n",
    "'Isocitrate dehydrogenase ',\n",
    "'catalyzes the oxidative decarboxylation of isocitrate',\n",
    "'1.1.1.42');\"\"\")\n",
    "\n",
    "\n",
    "\n",
    "#Pyruvate carboxylase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('10',\n",
    "'Pyruvate carboxylase',\n",
    "'carboxylation of pyruvate to form oxaloacetate',\n",
    "'6.4.1.1');\"\"\")\n",
    "\n",
    "\n",
    "#Phosphoenolpyruvate carboxylase(ppc)\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('11',\n",
    "'Phosphoenolpyruvate carboxylase',\n",
    "'Forms oxaloacetate, a four-carbon dicarboxylic acid source for the tricarboxylic acid cycle',\n",
    "'4.1.1.31');\"\"\")\n",
    "\n",
    "\n",
    "\n",
    "#G-6P dehyrogenase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('12',\n",
    "'Glucose-6-phosphate 1-dehydrogenase',\n",
    "'Convert Glucose-6-Phosphate to 6-phosphogluconate',\n",
    "'1.1.1.49');\"\"\")\n",
    "\n",
    "\n",
    "#Phosphoglucomutase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('13',\n",
    "'Phosphoglucomutase',\n",
    "'Facilitates the interconversion of glucose 1-phosphate and glucose 6-phosphate',\n",
    "'5.4.2.2');\"\"\")\n",
    "\n",
    "\n",
    "\n",
    "#Transaldolase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('13',\n",
    "'Transaldolase',\n",
    "'Links the pentose phosphate pathway to glycolysis',\n",
    "'2.2.1.2');\"\"\")\n",
    "\n",
    "\n",
    "\n",
    "#Ribose 5-phosphate isomerase\n",
    "c.execute(\"\"\"INSERT INTO enzyme (id,\n",
    "       name,\n",
    "       function,\n",
    "       enzyme_commission_number)\n",
    "VALUES('14',\n",
    "'Ribose 5-phosphate isomerase',\n",
    "'Links the pentose phosphate pathway to glycolysis',\n",
    "'5.3.1.6');\"\"\")\n",
    "\n",
    "conn.commit\n",
    "c.execute(\"SELECT * from enzyme;\")\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())\n",
    "print(c.fetchone())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fetching the data \n",
    "Most of the data such as the chromosome which the gene is on is fecthed using the tool below. Putting out how do I get every single gene in separate cells will be too long and it lags my notebook(tried doing it in another notebook, loading the notebook itself and and running the cells take way too long). Thus I only put what I did for the Hexokinase gene below. Just changing what is the data I want in the term= argument changes the information fected."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Entrez.email = 'yinchen202009@berkeley.edu'\n",
    "handle = Entrez.esearch(db= 'nucleotide', \n",
    "                        term = 'Homo sapiens[ORGN] hexokinase', \n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'nucleotide', id = i, rettype = 'gb', retmode= 'text')\n",
    "    print(handle.read())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
